src/main/webapp/authentication-app/src/main-components/SignUpPage.js

```diff
@@ -0,0 +1,88 @@
+import React, { useState } from "react";
+import { Link } from "@reach/router";
+
+const SignUpPage = () => {
+    const [userEmail, setUserEmail] = useState("");
+    const [userPassword, setUserPassword] = useState("");
+    const [displayName, setDisplayName] = useState("");
+    const [error, setError] = useState(null);
+
+    const createUser = (event, userEmail, userPassword) => {
+        event.preventDefault();
+        setUserEmail("");
+        setUserPassword("");
+        setDisplayName("");
+    };
+
+    const onChangeHandler = (event) => {
+        const { name, value } = event.currentTarget;
+        if( name === userPassword) {
+            setUserPassword(value);
+        }
+        else if (name === displayName) {
+            setDisplayName(value);
+        }
+        else if (name === userEmail) {
+            setUserEmail(value);
+        }
+    };
```

Comment:
Why are these all going through a common change handler? Is there a reason you can't split them out into a DisplayNameChangehandler, PasswordChangeHandler, etc? Having to match the value a second time seems brittle.